.circleci/config.yml

```diff
@@ -71,14 +72,19 @@ jobs:
 #            # Run sitespeed.io performance testing.
 #            mkdir ~/build/artifacts
 #            bash scripts/run-sitespeed-test.sh
-#      - save_cache:
-#          key: npm-v1-{{ checksum "package-lock.json" }}
-#          paths:
-#            - ~/build/node_modules
       - save_cache:
-          key: composer-v1-{{ checksum "composer.lock" }}
+          key: npm-v2-{{ checksum "gatsby/package-lock.json" }}
+          paths:
+            - ~/build/gatsby/node_modules
+      - save_cache:
+          key: composer-v2-{{ checksum "composer.lock" }}
           paths:
             - ~/.composer/cache
+      - save_cache:
+          key: v1-always-{{ epoch }}
```

Comment:
@alexfornuto I'm not sure what the benefit is of saving this key

> myapp-{{ epoch }} - Every build will generate separate cache keys.

From https://circleci.com/docs/2.0/caching/

If a cache is never used, why make it?

Reply:
^ I think Steve is correct here. You want a less random cache key, otherwise every time you trigger a re-build it's going to generate a new cache.

Reply:
My understanding here is that the restore line [here](https://github.com/pantheon-systems/documentation/pull/5004/files#diff-1d37e48f9ceff6d8030570cd36286a61R11) will pull any cache that matches the format `v1-always-`

@felicianotech would you care to weigh in?

Reply:
You can see [here](https://circleci.com/gh/pantheon-systems/documentation/14130) that the cache *is* being used.

Reply:
So in theory it'll just grab the newest cache always? I suppose that makes sense!

Reply:
To clarify: It will definitely create a new cache, but it will use the new cache each time on the next build - I figured this would keep the cache updated with new content.

Reply:
P.S. I borrowed the configuration from [here[(https://github.com/gatsbyjs/gatsby/issues/7366#issuecomment-469433003)

Reply:
@alexfornuto Sorry, late to the party.

Yes, when it comes to cache loading, we do support partial matching on the key. I don't know your codebase well but not every package manager supports partial caches well. For some, a partial cache means it saves some of the work, which is great. For others, a partial cache is as good as no cache in which case you shouldn't do it.

If you're seeing cache hits here, and it's reducing the build time, then that sounds right to me.